File: android/app/src/main/java/com/mccontroller/net/Messages.kt
package com.mccontroller.net

/**
 * Control messages received from the PC server (TCP channel only).
 */
sealed class ControlMessage

data class HelloAckMsg(val status: Byte, val udpPort: Int) : ControlMessage()
data class StateChangeMsg(val mode: Byte) : ControlMessage()
data class PongMsg(val seq: Int) : ControlMessage()
data class UnknownMsg(val type: Byte, val payloadLength: Int) : ControlMessage()

File: ios/McController/Net/Messages.swift
import Foundation

/// Server-to-client messages decoded from the TCP control channel.
enum ControlMessage: Sendable, Equatable {
    case helloAck(status: UInt8, udpPort: UInt16)
    case stateChange(mode: UInt8)
    case pong(seq: UInt32)
    case probeAck(status: UInt8)
    case unknown(type: UInt8, payloadLength: Int)
}
